core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -247,7 +247,13 @@ private[spark] class ExecutorAllocationManager(
         }
       }
     }
-    executor.scheduleWithFixedDelay(scheduleTask, 0, intervalMillis, TimeUnit.MILLISECONDS)
+
+    // SPARK-32287: Avoid the immediate schedule() after ExecutorAllocationManager started in test
+    // so that we won't hit the race condition between thread "spark-dynamic-executor-allocation"
+    // and thread "pool-1-thread-1-ScalaTest-running".
+    val initialDelay = if (testing) conf.get(TEST_SCHEDULE_INTERVAL) else 0
```

Comment:
I guess this is ok for its current usage as really its just trying to keep it from firing at all. If someone really wanted to use the value then it seems like this could just cause tests to take longer.

perhaps we should just change this config to disabled the timer altogether?

Reply:
Yeah, I actually also tried that way. I'd like to disable the whole timer if you also think it's ok.